kernel/aster-nix/src/process/posix_thread/futex.rs

```diff
@@ -257,72 +268,83 @@ impl FutexBucket {
         max_nrequeues: usize,
     ) {
         let mut count = 0;
+        let mut item_cursor = self.items.front_mut();
+        while !item_cursor.is_null() {
+            if count >= max_nrequeues {
+                break;
+            }
 
-        self.queue.retain(|item| {
-            if count >= max_nrequeues || key != item.key {
-                true
-            } else {
-                let mut new_item = item.clone();
-                new_item.key = new_key;
-                another.enqueue_item(new_item);
+            if item_cursor.get().is_some_and(|item| item.key == key) {
+                let mut item = item_cursor.remove().unwrap();
+                item.key = new_key;
+                another.add_item(item);
                 count += 1;
-                false
+            } else {
+                item_cursor.move_next()
```

Comment:
Refactor the loop expression to make its structure more apparent.

```rust
while termination_conditions {
    if filter_condition1 { continue; }
    //..
    if filter_conditionN { continue; }

    // The main logic for process items
}
```

With this structure, the loop would now look like as follows.

```rust
while !item_cursor.is_null() && count < max_nrequeues {
    if let Some(item) = item_cursor.get() else {
        item_cursor.move_next();
        continue;
    }
    if item.key != key {
        continue;
    }

    let mut item = item_cursor.remove().unwrap();
    item.key = new_key;
    another.add_item(item);

    count += 1; 
}
```

This new structure is more clean because (1) all termination points are centralized; (2) all filter conditions are written in a similar pattern; (3) the main logic is put in a more prominent place, without any indentation. Use this pattern to rewrite all loops in this file.